### PR TITLE
Delete obsolete documentation

### PR DESCRIPTION
**Pull request by MichaelSimons**
First pass at deleting documentation that is obsolete.  This will help avoid confusion as this documentation pollutes the repo and can be misleading.

### CODE REVIEW

File: Documentation/EngineeringPlanLinuxDistroRepos.md
Comment (mthalman) [inline Documentation/EngineeringPlanLinuxDistroRepos.md:1]: While most of this content is obsolete, I think there are some parts worth keeping such as "Toolset for Building .NET Core" and "N-1 Toolset".

File: Documentation/RepoApi.md [hunk 1]
```diff
@@ -1,109 +0,0 @@
...
-
-### Parameters
-
-Each action above will take a set of parameters used to pass information into the action and control its behavior.
-
-Parameters will take the form of `-ParameterName ParameterValue`. This form avoids any confusion with file paths on Unix machines by not using `/` to start the argument and works well in both powershell and shell script parsing. Every parameter passed by source-build will have a value, even if that value is just `true`.
-
-In order to avoid conflicts between existing properties (MSBuild or otherwise) used by the repos and the properties passed in by source-build, source-build will append a prefix to every property name it passes in: `DotNet`.
-
-#### Detecting build from source
```
[the remark this answers is not in the document]
Comment (MichaelSimons) [reply Documentation/RepoApi.md:43]: This is covered byt the [sourcebuild-in-repos documentation](https://github.com/dotnet/source-build/tree/main/Documentation/sourcebuild-in-repos).  Specifically [here](https://github.com/dotnet/source-build/blob/main/Documentation/sourcebuild-in-repos/new-repo.md#excluding-components)

File: Documentation/EngineeringPlanLinuxDistroRepos.md
Comment (MichaelSimons) [reply Documentation/EngineeringPlanLinuxDistroRepos.md:1]: The "N-1 Toolset" is covered by https://github.com/dotnet/source-build/blob/main/Documentation/bootstrapping-guidelines.md#building-a-servicing-release-of-net.  Specifically the n-1 use in servicing is addressed in https://github.com/dotnet/source-build/blob/main/Documentation/bootstrapping-guidelines.md#building-a-servicing-release-of-net and the need to rebootstrap in new major builds is covered in the opening.

I agree the "Toolset for Building.NET Core" does have some content that should be preserved.  Thinking about where to put this...

Comment (MichaelSimons) [reply Documentation/EngineeringPlanLinuxDistroRepos.md:1]: I preserved the distro packaging guideline exception in https://github.com/dotnet/source-build/pull/3605/commits/a7811f4ea1470833bd287f1ca71a218435ec2e91

**Review by premun — APPROVED** [review]

File: Documentation/EngineeringPlanLinuxDistroRepos.md
Comment (mthalman) [reply Documentation/EngineeringPlanLinuxDistroRepos.md:1]: > I agree the "Toolset for Building.NET Core" does have some content that should be preserved. Thinking about where to put this...

Is the plan to update this PR to include that or do that in a separate PR?

**Review by mthalman — APPROVED** [review]

**Pull request merged**